src/main/controllers/CrownWarnedListController.ts

```diff
@@ -0,0 +1,55 @@
+import {PipRequest} from '../models/request/PipRequest';
+import {Response} from 'express';
+import {cloneDeep} from 'lodash';
+import {PublicationService} from '../service/publicationService';
+import {DataManipulationService} from '../service/dataManipulationService';
+import {CrownWarnedListService} from '../service/listManipulation/crownWarnedListService';
+
+const publicationService = new PublicationService();
+const dataManipulationService = new DataManipulationService();
+const crownWarnedListService = new CrownWarnedListService();
+
+const listUrl = 'crown-warned-list';
+const toBeAllocated = 'To be allocated';
+
+export default class CrownWarnedListController {
+  public async get(req: PipRequest, res: Response): Promise<void> {
+    const artefactId = req.query.artefactId as string;
+    const searchResults = await publicationService.getIndividualPublicationJson(artefactId, req.user?.['piUserId']);
+    const metaData = await publicationService.getIndividualPublicationMetadata(artefactId, req.user?.['piUserId']);
+
+    if (searchResults && metaData) {
+      const publishedTime = dataManipulationService.publicationTimeInBst(searchResults['document']['publicationDate']);
+      const publishedDate = dataManipulationService.publicationDateInBst(searchResults['document']['publicationDate']);
+      const pageLanguage = publicationService.languageToLoadPageIn(metaData.language, req.lng);
+
+      const listData = crownWarnedListService.manipulateData(JSON.stringify(searchResults));
+      const toBeAllocatedData = [];
+      // Pull out 'to be allocated' data from the list data so it can be placed in a different section in teh templating engine
+      listData.forEach((value, key) => {
```

Comment:
You could replace a lot of these lines with filter or reduce instead of a forEach if you wanted to go more cool and functional.

Something along the lines of:
const unallocated = listData.filter(row -> row.courtRoom.toLowerCase().includes('to be allocated')
const allocated = listData.filter(row -> !unallocated.includes(row))

Be warned though - is your unallocated data going to be sorted by date?

Reply:
I have updated this part of the unallocated data is sorted to the end of the Map :)